### ISSUE TITLE
Zapier integration missing username of person who last replied

### ISSUE THREAD

**Issue by jackmcconnell** [WPChill/kb-support#192]
Using the Zapier integration, there's no way to determine who the last user was that replied to a ticket when a ticket is updated and that information is sent to Zapier. 
This is very important as currently, when anyone replies to a ticket - a customer, or a support agent, a message is sent to Zapier but there's no way within Zapier, when constructing a Zap, to be able to filter out who that reply came from. As such, the zap triggers on both replies from customers and support agents. 
Ideally, the information on what username or ID should be included along with the information sent to Zapier so that it can be filtered later on.

**Comment by jackmcconnell** [comment]
Also noticed today that for every new ticket that is created in the KB Support plugin on my site, it is sending two messages to Zapier - one for when a new ticket is created, and another immediately afterwards for a ticket update even though there have been no subsequent updates by the customer. 
I think this may be a bug where an update is occurring after a publish action on the ticket (possibly to save additional meta data) but both are being sent to Zapier.

**Comment by jackmcconnell** [comment]
A bigger issue is that updates to tickets aren't sending a message to Zapier. As such, I'm not being alerted of new tickets :/

**Comment by razvanaldea89** [comment]
@jackmcconnell I am a bit confused regarding your last comment. Trying to understand if the system is sending too many messages to Zapier or not sending any

**Comment by jackmcconnell** [comment]
So there are two main issues i've found in my testing: 

1. Two messages are being sent to Zapier when a new ticket is created in KB Support - one for the initial ticket published, and then another immediately after as an update. There should only be one - ideally the one containing the most amount of information.
2. When a new ticket reply is added, it isn't sending a message to Zapier. Presumably this is because the message that's sent to Zapier is for ticket updates, not ticket replies, which is a different post type.

**Comment by razvanaldea89** [comment]
Yes, seems like the Zapier extension is missing the action for new reply. For the moment you can hook it to `kbs_reply_to_ticket`  found [here](https://github.com/WPChill/kb-support/blob/master/includes/tickets/class-kbs-ticket.php#L2083) action if you want to roll your own solution for replies.  As for the ticket update problem, you can hook to `kbs_zapier_log_ticket_updated_data` filter that gives you 2 parameters, an empty array and the ticket ID,  to check what has changed and empty the data if it ain't something important ( Zapier action will not get triggered if there is no data ) ( will make a issue on our extension repo and a dev team member will look into it )

**Comment by jackmcconnell** [comment]
@razvanaldea89 Thanks. I've now gave something working for the ticket replies in Zapier.
